content/en/docs/languages/net/instrumentation.md

```diff
@@ -63,320 +279,362 @@ using var tracerProvider = Sdk.CreateTracerProviderBuilder()
+using Microsoft.AspNetCore.Mvc;
+using System.Diagnostics;
+using System.Net;
 
-```csharp
-public static void ParentOperation()
+public class DiceController : ControllerBase
 {
-    using var parentActivity = MyActivitySource.StartActivity("ParentActivity");
+    private ILogger<DiceController> logger;
+    
+    public static readonly ActivitySource MyActivitySource = new("Dice.Server", "1.0.0");
 
-    // Do some work tracked by parentActivity
-
-    ChildOperation();
+    public DiceController(ILogger<DiceController> logger)
+    {
+        this.logger = logger;
+    }
 
-    // Finish up work tracked by parentActivity again
+    //...
 }
 
-public static void ChildOperation()
-{
-    using var childActivity = MyActivitySource.StartActivity("ChildActivity");
 
-    // Track work in ChildOperation with childActivity
-}
 ```
 
-When you view spans in a trace visualization tool, `ChildActivity` will be
-tracked as a nested operation under `ParentActivity`.
+And second, in the library file `Dice.cs`:
 
-### Nested Activities in the same scope
+```csharp
 
-You may wish to create a parent-child relationship in the same scope. Although
-possible, this is generally not recommended because you need to be careful to
-end any nested `Activity` when you expect it to end.
+using System.Diagnostics;
 
-```csharp
-public static void DoWork()
+public class Dice
 {
-    using var parentActivity = MyActivitySource.StartActivity("ParentActivity");
+    public static readonly ActivitySource MyActivitySource = new("Dice.Lib", "1.0.0");
+    
+    private int min;
+    private int max;
 
-    // Do some work tracked by parentActivity
-
-    using (var childActivity = MyActivitySource.StartActivity("ChildActivity"))
+    public Dice(int min, int max)
     {
-        // Do some "child" work in the same function
+        this.min = min;
+        this.max = max;
     }
 
-    // Finish up work tracked by parentActivity again
+    //...
 }
+
 ```
 
-In the preceding example, `childActivity` is ended because the scope of the
-`using` block is explicitly defined, rather than scoped to `DoWork` itself like
-`parentActivity`.
+### Create Activities
 
-### Creating independent Activities
+Now that you have [activitySources](/docs/concepts/signals/traces/#tracer) initialized, you can create [activities](/docs/concepts/signals/traces/#spans).
 
-The previous examples showed how to create Activities that follow a nested
-hierarchy. In some cases, you'll want to create independent Activities that are
-siblings of the same root rather than being nested.
+The code below illustrates how to create an activity.
 
 ```csharp
-public static void DoWork()
+public List<int> rollTheDice(int rolls)
 {
-    using var parent = MyActivitySource.StartActivity("parent");
-
-    using (var child1 = DemoSource.StartActivity("child1"))
-    {
-        // Do some work that 'child1' tracks
-    }
+    var activity = MyActivitySource.StartActivity("rollTheDice");
```

Comment:
I'd also suggest to add a note here saying something like:
The example here is creating a new Span for demonstration purposes only. It is recommended to create Spans, only if it is doing something worth measuring independently. Too many spans makes it harder to visualize in tools like Jaeger.

https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/examples/AspNetCore/Controllers/WeatherForecastController.cs#L45-L52 - These wording maybe borrowed.